testutil/keeper/unit_test_helpers.go

```diff
@@ -22,169 +23,173 @@ import (
 	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
 	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
 	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v3/modules/core/23-commitment/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v3/modules/light-clients/07-tendermint/types"
+	providertypes "github.com/cosmos/interchain-security/x/ccv/provider/types"
 )
 
-// Constructs a provider keeper and context object for unit tests, backed by an in-memory db.
-func GetProviderKeeperAndCtx(t testing.TB) (providerkeeper.Keeper, sdk.Context) {
+// Parameters needed to instantiate an in-memory keeper
+type InMemKeeperParams struct {
+	Cdc            *codec.ProtoCodec
+	StoreKey       *storetypes.KVStoreKey
+	ParamsSubspace *paramstypes.Subspace
+	Ctx            sdk.Context
+}
 
-	cdc, storeKey, paramsSubspace, ctx := SetupInMemKeeper(t)
+// NewInMemKeeperParams instantiates in-memory keeper params with default values
+func NewInMemKeeperParams(t testing.TB) InMemKeeperParams {
+	storeKey := sdk.NewKVStoreKey(types.StoreKey)
+	memStoreKey := storetypes.NewMemoryStoreKey(types.MemStoreKey)
+
+	db := tmdb.NewMemDB()
+	stateStore := store.NewCommitMultiStore(db)
+	stateStore.MountStoreWithDB(storeKey, sdk.StoreTypeIAVL, db)
+	stateStore.MountStoreWithDB(memStoreKey, sdk.StoreTypeMemory, nil)
+	require.NoError(t, stateStore.LoadLatestVersion())
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
 
-	k := providerkeeper.NewKeeper(
-		cdc,
+	paramsSubspace := paramstypes.NewSubspace(cdc,
+		codec.NewLegacyAmino(),
 		storeKey,
-		paramsSubspace,
-		&MockScopedKeeper{},
-		&MockChannelKeeper{},
-		&MockPortKeeper{},
-		&MockConnectionKeeper{},
-		&MockClientKeeper{},
-		&MockStakingKeeper{},
-		&MockSlashingKeeper{},
-		&MockAccountKeeper{},
-		"",
+		memStoreKey,
+		paramstypes.ModuleName,
 	)
-	return k, ctx
-}
+	ctx := sdk.NewContext(stateStore, tmproto.Header{}, false, log.NewNopLogger())
```

Comment:
I think it would be useful to have the ctx creation be a separate function, taking stateStore, as argument. It may also be useful to be able to customize the header a bit (but we can leave that for a later PR). 

Then it would be good to make the user call the two parts separately

```
params := NewInMemKeeperParams(t)
ctx := NewCtx(params.statestore ) // would need to add handle to this to params
```

Maybe I'm overthinking it. I need to double check how ctx works. The thing is you want one ctx for the whole app (for a given block) AFAIK. So not sure what the effect is of having a different ctx for each keeper. It might be limiting.

Reply:
Useful input here, this is making me really think through my design!

I don't really see a use case in which a user would want to pass in their own ```params.statestore``` to a context object. They'd pretty much always be happy with the default in-mem store, if they're writing pure unit tests. More customization should be reserved for other forms of testing imo. I could be missing a use-case tho, maybe we just make this change in the future if it becomes needed?

On customizing the ctx header, etc. after it has been constructed, this is already possible. The caller just needs to use
```go
ctx = ctx.WithBlockHeader(tmproto.Header{})
ctx = ctx.WithBlockTime(time.Now())
...etc
``` 

Emphasis with the design of this PR is the ability to setup a unit test with minimal lines of code, but still allow a user to replace default values for things, _only if needed_

Reply:
Hmm ok, fair point! I agree we can just change it if needed. YAGNI lol